fluent-bundle/src/resolve.rs

```diff
@@ -141,8 +125,36 @@ pub fn resolve_value_for_entry<'source>(
     FluentValue::String(string.into())
 }
 
+fn generate_ref_error<'source, R: Borrow<FluentResource>>(
+    env: &mut Scope<'source, R>,
+    node: DisplayableNode<'source>,
+) -> FluentValue<'source> {
+    env.errors.push(ResolverError::Reference(node.get_error()));
+    FluentValue::Error(node)
+}
+
+// Converts an AST node to a `FluentValue`.
+pub trait ResolveValue<'source> {
+    fn resolve<R: Borrow<FluentResource>>(
+        &self,
+        env: &mut Scope<'source, R>,
+    ) -> FluentValue<'source>;
```

Comment:
Perhaps a nicer way to express this would be with a `where` clause?

```rust
pub trait ResolveValue<'source> {
    fn resolve<R>(&self, env: &mut Scope<'source, R>) -> FluentValue<'source>
    where
        R: Borrow<FluentResource>;
}
```

And same with all the `impl`s.